dask/dataframe/core.py

```diff
@@ -4204,17 +4204,37 @@ def _take_last(a, skipna=True):
         Whether to exclude NaN
 
     """
+
+    def _collect_series_last_valid(s):
+        size = len(s)
+        for i in range(size - 1, max(size - 11, 0) - 1, -1):
```

Comment:
I recommend `range(0, min(10, len(s)))` instead and use `-i` below.  I suspect that this will be easier for future reviewers to follow than the negative logic here.

Reply:
Agreed.

Reply:
Starting range with `1` as otherwise we would grab the first row in the column unintentionally.